### ISSUE TITLE
Error de validación: bloqueo funcional e incoherencia de interfaz

### ISSUE THREAD

**Issue by juacas** [UNIMOODLE/moodle-mod_certifygen#82]
Escenario:
1) Entrar como administrador.
2) Emitir manualmente un certificado para un estudiante.
3) La petición se marca como "Error de validación", pero no se puede hacer nada en el mismo interfaz.
![image](https://github.com/user-attachments/assets/df6079b5-9340-4023-a939-d1a2c2e2b011)
4) En la pantalla de errores de Certifygen se ve:
![image](https://github.com/user-attachments/assets/6fb49ece-1a58-46e8-a5b3-afa0c350e750)
Por lo que debe haber habido un error temporal de comunicación con el servidor.
5) Entro como un estudiante: El estudiante en su vista ve el botón de emitir certificado, no ve el error.
6) El estudiante pide el certificado y tras un ciclo de CRON lo obtiene y lo puede descargar.
![image](https://github.com/user-attachments/assets/c5647613-d018-4d5c-8e64-438e9f7c740a)
![image](https://github.com/user-attachments/assets/ecf98d31-4561-4ae3-9644-e92d70872cab)
7) El administrador en su listado ve algo incoherente: estado Not Started.
![image](https://github.com/user-attachments/assets/af1ae3b6-b361-48aa-b065-6c6f11a280b1)

Errores a comprobar/corregir:
1) El administrador (o usuario con permisos) debe poder reconducir un error en el interfaz de usuario. En este caso no hay botones de acción, por ejemplo para reintentar.
2) El estado de ambas vistas es incompatible, algo pasa que puede ser indicio de otro problema más serio.

Tablas que considero relevantes:

mdl_certifygen_error
---
| id | validationid | status | code | message | usermodified | timecreated | timemodified | 
| ---: | ---: | ---: | --- | --- | ---: | ---: | ---: | 
| 1 | 5 | 4 | validation_plugin_send_file_error | Empty reply from server | 2 | 1731919083 | 1731919083 | 

mdl_certifygen_validationcsv
---
| id | validationid | applicationid | token | usermodified | timecreated | timemodified | 
| ---: | ---: | --- | --- | ---: | ---: | ---: | 
| 1 | 1 | FIRMA-DOCUMENTAL-2024-01191 | 4194164710KA | 2 | 1729084684 | 1729084684 | 
| 2 | 2 | FIRMA-DOCUMENTAL-2024-01192 | 6719190900KA | 2 | 1729084722 | 1729084722 | 
| 3 | 3 | FIRMA-DOCUMENTAL-2024-01193 | 3062734292VC | 2 | 1729084863 | 1729084863 | 
| 4 | 4 | FIRMA-DOCUMENTAL-2024-01194 | TR0881280479AU | 2 | 1729085077 | 1729085077 | 
| 5 | 6 | FIRMA-DOCUMENTAL-2024-01221 | 8334766023BP | 2 | 1731919504 | 1731919504 | 
| 6 | 7 | FIRMA-DOCUMENTAL-2024-01222 | 8299339383TS | 2942 | 1731920570 | 1731920570 | 

mdl_certifygen_validations
---
| id | name | courses | code | certifygenid | issueid | userid | modelid | lang | status | usermodified | timecreated | timemodified | 
| ---: | --- | --- | --- | ---: | ---: | ---: | ---: | --- | ---: | ---: | ---: | ---: | 
| 1 | \N | \N | \N | 1 | 16 | 2963 | 1 | en | 8 | 2 | 1729084670 | 1729085005 | 
| 2 | \N | \N | \N | 1 | 17 | 2963 | 1 | es | 8 | 2 | 1729084710 | 1729085006 | 
| 3 | \N | \N | \N | 1 | 18 | 2948 | 1 | en | 8 | 2 | 1729084851 | 1729085007 | 
| 4 | test1 | 6,28,30 | TR0881280479AU | 0 | \N | 2 | 2 | en | 8 | 2 | 1729085062 | 1729085079 | 
| 5 | \N | \N | \N | 1 | 19 | 2942 | 1 | en | 1 | 2 | 1731919080 | 1731920615 | 
| 6 | \N | \N | \N | 1 | 20 | 381 | 1 | en | 8 | 2 | 1731919483 | 1731920668 | 
| 7 | \N | \N | \N | 1 | 21 | 2942 | 1 | es | 8 | 2 | 1731920555 | 1731920668 |

**Comment by elena3ip** [comment]
El idioma con el que emite el administrador tambien es español?

**Comment by elena3ip** [comment]
Si lo pantallazos de la base de datos corresponden a esta actividad se ve en la tabla mdl_certifygen_validations con id=6 que se ha intentado emitir en ingles. Por eso al alumno en español no le sale el error.

**Comment by juacas** [comment]
No, ha debido ser un fallo transitorio, porque el resto de intentos han funcionado bien.
Afortunadamente ha ocurrido y podremos revisar las comprobaciones del estado que se ha generado, porque me extraña que se genere un bloqueo en el interfaz de usuario del administrador y que el estudiante pueda emitir el certificado.

**Comment by juacas** [comment]
El comentario sobre el idioma y el listado se relaciona con #10 y va a ser un problema serio de incidencias en producción.  Parece poca cosa pero es grave.
